R/utilities-sensitivity-analysis.R

```diff
@@ -1,342 +1,343 @@
-
-
-  outputColumn <- NULL
-  pkParameterColumn <- NULL
-  individualIdColumn <- NULL
-  valuesColumn <- NULL
-  unitsColumn <- NULL
-  quantileColumn <- NULL
-
-  for (output in outputs) {
-    for (pkParameter in pkParameters) {
-      singleOuputSinglePKDataframe <- allPKResultsDataframe[ allPKResultsDataframe["QuantityPath"] == output & allPKResultsDataframe["Parameter"] == pkParameter, ]
-      quantileResults <- getQuantileIndividualIds(singleOuputSinglePKDataframe, quantileVec)
-      for (n in 1:length(quantileResults$ids)) {
-        outputColumn <- c(outputColumn, output)
-        pkParameterColumn <- c(pkParameterColumn, pkParameter)
-        individualIdColumn <- c(individualIdColumn, quantileResults$ids[n])
-        valuesColumn <- c(valuesColumn, quantileResults$values[n])
-        unitsColumn <- c(unitsColumn, quantileResults$units[n])
-        quantileColumn <- c(quantileColumn, quantileVec[n])
-      }
-    }
-  }
-
-  filenamesColumn <-  paste0(sapply(X = individualIdColumn,FUN = getIndividualSAResultsFileName,resultsFileName),".csv")
-  sensitivityAnalysesResultsIndexFileDF <- data.frame("Outputs" = outputColumn, "pkParameters" = pkParameterColumn, "Quantile" = quantileColumn, "Value" = valuesColumn, "Unit" = unitsColumn,  "IndividualId" = individualIdColumn, "Filename" = filenamesColumn)
-  write.csv(x = sensitivityAnalysesResultsIndexFileDF, file = file.path(resultsFileFolder, paste0(popSAResultsIndexFile, ".csv")))
-  return(sensitivityAnalysesResultsIndexFileDF)
-}
-
-
-#' @title getIndividualSAResultsFileName
-#' @description Function to build name of inidividual SA results file
-#' @param resultsFileName root name of population sensitivity analysis results CSV files
-#' @param individualId id of individual
-getIndividualSAResultsFileName <- function(individualId,resultsFileName){
-  return(paste(resultsFileName,"IndividualId",individualId,sep="-"))
-}
+#' @title analyzeSensitivity
+#' @description Determine whether to run SA for individual or population.  If for individual,  pass simulation to individualSensitivityAnalysis.
+#' If SA is for population, loop thru population file, extract parameters for each individual, and pass them to individualSensitivityAnalysis.
+#' @param simFilePath path to simulation file
+#' @param parametersToPerturb paths to parameters to perturb in sensitivity analysis
+#' @param popFilePath path to the population data file
+#' @param individualID ID of individual in population data file for whom to perform sensitivity analysis
+#' @param numberOfCores number of cores over which to parallelize the sensitivity analysis
+#' @param resultsFileFolder path to population sensitivity analysis results CSV files
+#' @param resultsFileName root name of population sensitivity analysis results CSV files
+#' @return SA results for individual or population
+#' @export
+#' @import ospsuite
+analyzeSensitivity <- function(simFilePath,
```

Comment:
I am bit confused with the terminology
We are dealing with `SensitivityAnalysis`
Here you analyze the sentisivity analysis. But in fact you are running the sensitivity analysis.? 
Shouldn't we call it runSensitivity and that would be the only exported method? All ther other one private?

Reply:
Renamed it to runSensitivity

populationSensitivityAnalysis is another function in this file that needs to be exported because it is accessed by the workflow class.

I had kept a few other methods exported in case user needs quick & more direct access for very specific functions.  eg 
1 - individualSensitivityAnalysis is used to perform SA for one individual (ie for a specific individual ID, with individual parameters coming from a population ), possibly in parallel
2 - analyzeCoreSensitivity is used to perform SA for an given simulation on one core.  Mainly used for the parallel runs.  

removing export for these latter two for now

Reply:
actually we need analyzeCoreSensitivity exported for access by cores